ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/ParseProxyTest.java

```diff
@@ -59,6 +63,10 @@ public void testOutput() {
             + "{\"submission_timestamp\":\"2000-01-01T00:00:00.000000Z\"" //
             + ",\"x_forwarded_for\":\"3, 2, 1\"" //
             + "},\"payload\":\"notProxied++\"}",
+        "{\"attributeMap\":" //
+            + "{\"submission_timestamp\":\"2000-01-01T00:00:00.000000Z\"" //
+            + ",\"x_forwarded_for\":\"4, 3, 6.7.8.9\"" //
```

Comment:
Note that the logic as written currently in this PR is attempting to follow exactly what's written in https://bugzilla.mozilla.org/show_bug.cgi?id=1729069#c8; itchecks for the `proxyIpAddress`, and if it exists strips the _trailing_ entry from the IP. This leads to the static IP still existing in X-Forwarded-For, although the later IP lookups only care about the second-to-last entry, so this works.

It would make the most sense to me to modify `ParseProxy` to parse X-Forwarded-For into an array of IPs, check each entry against `proxyIpAddress`. If we find a match, we remove it, and then `mkString` the array back into a comma-separated list. In that case, this test case would change to `4,3,1` rather than `4, 3, 6.7.8.9`.

Reply:
I agree that parsing this into a list and removing `proxyIpAddress` if extant makes the most sense. I wrote the logic in the comment to be as similar to what we already had as possible, but this should achieve the same effect more clearly.